### PR TITLE
[Php80] Add AddParamBasedOnParentClassMethodRector

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes https://github.com/rectorphp/rector/issues/6851

### CODE REVIEW

File: rules/Php80/Rector/ClassMethod/AddParamBasedOnParentClassMethodRector.php [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+        if ($parentClassMethod->isPrivate()) {
+            return null;
+        }
+
+        $currentClassMethodParams = $node->getParams();
+        $parentClassMethodParams  = $parentClassMethod->getParams();
+
+        if (count($currentClassMethodParams) < count($parentClassMethodParams)) {
+            $node->params = $parentClassMethodParams;
```
Comment (samsonasik) [inline rules/Php80/Rector/ClassMethod/AddParamBasedOnParentClassMethodRector.php]: replacing current classmethod params with parent classmethod params seems got error:

```bash
There was 1 failure:

1) Rector\Tests\Php80\Rector\ClassMethod\AddParamBasedOnParentClassMethodRector\AddParamBasedOnParentClassMethodRectorTest::test with data set #0 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
assert($startPos >= 0 && $endPos >= 0) in /Users/samsonasik/www/rector-src/vendor/nikic/php-parser/lib/PhpParser/PrettyPrinterAbstract.php:535
```

need to find different solution.

Comment (samsonasik) [reply rules/Php80/Rector/ClassMethod/AddParamBasedOnParentClassMethodRector.php]: Fixed by loop parent params and re-create params with new Variable with empty attributes https://github.com/rectorphp/rector-src/pull/1431/commits/4b3bec8b5d5e6737e7bd3b7c7db688cde93b6a87

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Review by TomasVotruba — APPROVED** [review]

**Pull request merged**